scr/main/java/bittercode/model/User.java

```diff
@@ -0,0 +1,90 @@
+package com.bittercode.model;
+
+import java.io.Serializable;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+public class User implements Serializable {
+    private String emailId;
+    private String password;
+    private String firstName;
+    private String lastName;
+    private Long phone;
+    private String address;
+    private List<UserRole> roles;
+
+    public String getEmailId() {
+        return emailId;
+    }
+
+    public void setEmailId(String emailId) {
+        this.emailId = emailId;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    public Long getPhone() {
+        return phone;
+    }
+
+    public void setPhone(Long phone) {
+        this.phone = phone;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public List<UserRole> getRoles() {
+        return roles;
+    }
+
+    public void setRoles(List<UserRole> roles) {
+        this.roles = roles;
+    }
+
+    public static User retrieveFromHttpServletRequest(HttpServletRequest req) {
+        User user = new User();
+        String pWord = req.getParameter(UsersDBConstants.COLUMN_PASSWORD);
+        String fName = req.getParameter(UsersDBConstants.COLUMN_FIRSTNAME);
+        String lName = req.getParameter(UsersDBConstants.COLUMN_LASTNAME);
+        String addr = req.getParameter(UsersDBConstants.COLUMN_ADDRESS);
+        String phNo = req.getParameter(UsersDBConstants.COLUMN_PHONE);
+        String mailId = req.getParameter(UsersDBConstants.COLUMN_MAILID);
+
+        user.setEmailId(mailId);
+        user.setFirstName(fName);
+        user.setLastName(lName);
+        user.setPassword(pWord);
+        user.setPhone(Long.parseLong(phNo));
+        user.setAddress(addr);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Add phone number validation check</b></div>
<div id="fix">

Consider adding validation for the phone number string before parsing to avoid potential `NumberFormatException`. The `phNo` parameter could be null or non-numeric.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -84,7 +84,11 @@
         user.setLastName(lName);
         user.setPassword(pWord);
 -        user.setPhone(Long.parseLong(phNo));
 +        if (phNo != null && phNo.matches("\\d+")) {
 +            user.setPhone(Long.parseLong(phNo));
 +        } else {
 +            throw new IllegalArgumentException("Invalid phone number format");
 +        }
         user.setAddress(addr);
```


```suggestion
       user.setLastName(lName);
       user.setPassword(pWord);
        if (phNo != null && phNo.matches("\\d+")) {
            user.setPhone(Long.parseLong(phNo));
        } else {
            throw new IllegalArgumentException("Invalid phone number format");
        }
       user.setAddress(addr);
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/7#issuecomment-2527740799>#4862f0</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged